diff --git a/src/module_media.pri b/src/module_media.pri
index 4c5f9ef..9ac3086 100644
--- a/src/module_media.pri
+++ b/src/module_media.pri
@@ -75,8 +75,6 @@ enable_qtopiamedia:contains(QTOPIAMEDIA_ENGINES,cruxus):!enable_singleexec {
     # MP3 support for cruxus
     # not in source packages
     PROJECTS*=\
-        3rdparty/libraries/libmad\
-        3rdparty/plugins/codecs/libmad \
         3rdparty/libraries/amr\
         3rdparty/plugins/codecs/libamr
 }
