diff --git a/qtopiacore/qt/src/gui/widgets/qlineedit.cpp b/qtopiacore/qt/src/gui/widgets/qlineedit.cpp
index 70c7d5c..2aea5b7 100644
--- a/qtopiacore/qt/src/gui/widgets/qlineedit.cpp
+++ b/qtopiacore/qt/src/gui/widgets/qlineedit.cpp
@@ -98,7 +98,7 @@ extern void qt_mac_secure_keyboard(bool); //qapplication_mac.cpp
 static inline bool shouldEnableInputMethod(QLineEdit *lineedit)
 {
     const QLineEdit::EchoMode mode = lineedit->echoMode();
-    return !lineedit->isReadOnly() && (mode == QLineEdit::Normal || mode == QLineEdit::PasswordEchoOnEdit);
+    return !lineedit->isReadOnly();// && (mode == QLineEdit::Normal || mode == QLineEdit::PasswordEchoOnEdit);
 }
 
 /*!
