Index: qtmoko/devices/gta04/mkspecs/qws/linux-native-g++/qmake.conf
===================================================================
--- qtmoko.orig/devices/gta04/mkspecs/qws/linux-native-g++/qmake.conf	2012-10-10 07:48:37.700164936 +0200
+++ qtmoko/devices/gta04/mkspecs/qws/linux-native-g++/qmake.conf	2012-10-10 07:50:58.804854752 +0200
@@ -8,6 +8,10 @@
 QT                      += core gui network
 QMAKE_INCREMENTAL_STYLE = sublib
 
+QMAKE_PREFIX_SHLIB      = lib
+QMAKE_PREFIX_STATICLIB  = lib
+QMAKE_EXTENSION_STATICLIB = a
+
 QMAKE_CC		= gcc
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		=
Index: qtmoko/devices/neo/mkspecs/qws/linux-native-g++/qmake.conf
===================================================================
--- qtmoko.orig/devices/neo/mkspecs/qws/linux-native-g++/qmake.conf	2012-10-10 07:48:22.856092356 +0200
+++ qtmoko/devices/neo/mkspecs/qws/linux-native-g++/qmake.conf	2012-10-10 07:50:46.556794884 +0200
@@ -8,6 +8,10 @@
 QT                      += core gui network
 QMAKE_INCREMENTAL_STYLE = sublib
 
+QMAKE_PREFIX_SHLIB      = lib
+QMAKE_PREFIX_STATICLIB  = lib
+QMAKE_EXTENSION_STATICLIB = a
+
 QMAKE_CC		= gcc
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		=
