Index: qtmoko/qtopiacore/qt/src/tools/bootstrap/bootstrap.pro
===================================================================
--- qtmoko.orig/qtopiacore/qt/src/tools/bootstrap/bootstrap.pro	2012-09-23 16:22:07.321571253 +0200
+++ qtmoko/qtopiacore/qt/src/tools/bootstrap/bootstrap.pro	2012-09-25 22:02:40.030237968 +0200
@@ -96,7 +96,7 @@
    LIBS += -framework CoreServices
 }
 
-if(contains(QT_CONFIG, zlib)|cross_compile):include(../../3rdparty/zlib.pri)
+if(contains(QT_CONFIG, zlib)):include(../../3rdparty/zlib.pri)
 else:include(../../3rdparty/zlib_dependency.pri)
 
 lib.CONFIG = dummy_install
